audify/utils/api_config.py

```
@@ -43,6 +44,65 @@
+    (``reraise=True``) or a ``RuntimeError`` is raised with a clear message
+    naming the API that failed.
+
+    Only ``requests.RequestException`` (covers timeouts, connection errors,
+    HTTP errors raised via ``raise_for_status()``) and generic ``Exception``
+    from SDK clients (boto3, google-cloud, litellm) are caught.
+    """
+    last_exc: Optional[Exception] = None
+    for attempt in range(1, max_retries + 1):
+        try:
+            return func()
+        except Exception as exc:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)